src/test/java/org/sasanlabs/service/vulnerability/commandInjection/CommandInjectionTest.java

```diff
@@ -0,0 +1,117 @@
+    private static final String mockedSuccessPingResponse = "PING SUCCESS";
+
+    @BeforeEach
+    void setUp() throws IOException {
+        commandInjection = Mockito.spy(new CommandInjection());
+
+        StringBuilder stringBuilder = new StringBuilder().append(mockedSuccessPingResponse);
+
+        // Real localhost ping is not able to operate in CI, Mock instead
+        doReturn(stringBuilder)
+                .when(commandInjection)
+                .getResponseFromPingCommand(eq(LOCALHOST), eq(true));
+    }
+
+    @Test
+    public void getVulnerablePayloadLevel1_Localhost_ExpectPingSuccess() throws IOException {
+        // Act
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                commandInjection.getVulnerablePayloadLevel1(LOCALHOST);
+
+        // Assert
+        assertStatusAndValidate(responseEntity, LOCALHOST, mockedSuccessPingResponse);
+
+        verify(commandInjection).getVulnerablePayloadLevel1(LOCALHOST);
+    }
+
+    @Test
+    public void getVulnerablePayloadLevel1_NoIp_ExpectNoResponseContent() throws IOException {
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                commandInjection.getVulnerablePayloadLevel1(Strings.EMPTY);
+
+        // Assert
+        assertStatusAndValidate(responseEntity, Strings.EMPTY, Strings.EMPTY);
+
+        verify(commandInjection).getVulnerablePayloadLevel1(eq(Strings.EMPTY));
+    }
+
+    @Test
+    public void getVulnerablePayloadLevel2_Localhost_ExpectPingSuccess()
+            throws ServiceApplicationException, IOException {
+        RequestEntity entity = new RequestEntity(HttpMethod.GET, URI.create(LOCALHOST));
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                commandInjection.getVulnerablePayloadLevel2(LOCALHOST, entity);
+
+        assertStatusAndValidate(responseEntity, LOCALHOST, mockedSuccessPingResponse);
+
+        verify(commandInjection).getVulnerablePayloadLevel2(eq(LOCALHOST), eq(entity));
+    }
+
+    @Test
+    public void getVulnerablePayloadLevel2_NoIp_ExpectNoResponseContent()
+            throws ServiceApplicationException, IOException {
+        RequestEntity entity = new RequestEntity(HttpMethod.GET, URI.create(Strings.EMPTY));
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                commandInjection.getVulnerablePayloadLevel2(Strings.EMPTY, entity);
+
+        // Assert
+        assertStatusAndValidate(responseEntity, Strings.EMPTY, Strings.EMPTY);
+
+        verify(commandInjection).getVulnerablePayloadLevel2(eq(Strings.EMPTY), eq(entity));
+    }
+
+    @Test
+    public void getVulnerablePayloadLevel2_LocalhostAndSEMICOLON_ExpectNoResponseContent()
+            throws ServiceApplicationException, IOException {
+        RequestEntity entity =
+                new RequestEntity(HttpMethod.GET, URI.create("http://localhost:8080/;"));
```

Comment:
this seems like not a correct url. can you please start the application and try to inject a payload so that we can use the exact payload here.